js/src/main/scala/scalajson/ast/JValue.scala

```diff
@@ -77,22 +77,26 @@ object JNumber {
   def apply(value: Float): JValue = value match {
     case n if java.lang.Float.isNaN(n) => JNull
     case n if n.isInfinity => JNull
-    case _ => JNumber(value.toString)
+    case _ => new JNumber(value.toString)
   }
+
+  def fromString(value: String): Option[JNumber] =
```

Comment:
With Scala 2.10 out of the way we should be able to simply name this method `apply` as well, like all the others...